packages/create-invoice-form/src/lib/utils/prepareRequest.ts

```diff
@@ -18,105 +18,112 @@ export const prepareRequestParams = ({
   currency,
   formData,
   invoiceTotals,
-}: IRequestParams): Types.ICreateRequestParameters => ({
-  requestInfo: {
-    currency: {
-      type: currency.type,
-      value: currency.address,
-      network: currency.network,
-    },
-    expectedAmount: parseUnits(
-      invoiceTotals.totalAmount.toFixed(2),
-      currency.decimals
-    ).toString(),
-    payee: {
-      type: Types.Identity.TYPE.ETHEREUM_ADDRESS,
-      value: formData.creatorId,
-    },
-    payer: {
-      type: Types.Identity.TYPE.ETHEREUM_ADDRESS,
-      value: formData.payerAddress,
-    },
-    timestamp: Utils.getCurrentTimestampInSecond(),
-  },
-  paymentNetwork: {
-    id: Types.Extension.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT,
-    parameters: {
-      paymentNetworkName: currency.network,
-      paymentAddress: formData.payeeAddress,
-      feeAddress: zeroAddress,
-      feeAmount: "0",
-    },
-  },
-  contentData: {
-    meta: {
-      format: "rnf_invoice",
-      version: "0.0.3",
-    },
-    miscellaneous:
-      formData.miscellaneous.labels.length > 0
-        ? formData.miscellaneous
-        : undefined,
-    creationDate: new Date(formData.issuedOn).toISOString(),
-    invoiceNumber: formData.invoiceNumber,
-    note: formData.note.length > 0 ? formData.note : undefined,
-    invoiceItems: formData.items.map((item) => ({
-      name: item.description,
-      quantity: Number(item.quantity),
-      unitPrice: parseUnits(
-        item.unitPrice.toString(),
-        currency.decimals
-      ).toString(),
-      discount: parseUnits(
-        item.discount.toString(),
+}: IRequestParams): Types.ICreateRequestParameters => {
+  const isERC20 = currency.type === Types.RequestLogic.CURRENCY.ERC20;
+
+  return {
+    requestInfo: {
+      currency: {
+        type: currency.type,
+        value: isERC20 ? currency.address : "eth",
+        network: currency.network,
+      },
+      expectedAmount: parseUnits(
+        invoiceTotals.totalAmount.toFixed(2),
```

Comment:
We shouldn't be rounding with `toFixed(2)` when writing data from the user to the protocol. 

We should round only when displaying long numbers.

Let's create a new issue to fix this.

Reply:
https://github.com/RequestNetwork/web-components/issues/121